src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ValidateWorkflowsPass.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Workflow\Validator\DefinitionValidatorInterface;
+use Symfony\Component\Workflow\Validator\SinglePlaceWorkflowValidator;
+use Symfony\Component\Workflow\Validator\StateMachineValidator;
+use Symfony\Component\Workflow\Validator\WorkflowValidator;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ValidateWorkflowsPass implements CompilerPassInterface
+{
+    /**
+     * @var DefinitionValidatorInterface[]
+     */
+    private $validators = array();
+
+    public function process(ContainerBuilder $container)
+    {
+        $taggedServices = $container->findTaggedServiceIds('workflow.definition');
+        foreach ($taggedServices as $id => $tags) {
+            $definition = $container->get($id);
+            foreach ($tags as $tag) {
+                $this->getValidator($tag)->validate($definition, $tag['name']);
```

Comment:
Thank you. You are correct. I'll throw exception of some data is missing on the tag.